rmf_traffic/include/rmf_traffic/Trajectory.hpp

```diff
@@ -43,270 +42,115 @@ class Trajectory
+    ///   The new finishing time for this Trajectory Waypoint.
     ///
     /// \sa adjust_finish_times(Time new_time)
-    Segment& set_finish_time(Time new_time);
+    Waypoint& change_time(Time new_time);
 
-    /// Adjust the finishing time of this segment and all subsequent segments by
+    /// Adjust the finishing time of this waypoint and all subsequent segments by
     /// the given duration. This is guaranteed to maintain the ordering of the
-    /// Trajectory Segments, and is more efficient than changing all the times
+    /// Trajectory Waypoints, and is more efficient than changing all the times
     /// directly.
     ///
     /// \warning If a negative delta_t is given, it must not cause this
-    /// Segment's finish time to be less than or equal to the finish time of its
-    /// preceding Segment, or else a std::invalid_argument exception will be
+    /// Waypoint's finish time to be less than or equal to the finish time of its
+    /// preceding Waypoint, or else a std::invalid_argument exception will be
     /// thrown.
     ///
     /// \param[in] delta_t
-    ///   How much to change the finishing time of this segment and all later
+    ///   How much to change the finishing time of this waypoint and all later
     ///   segments. If negative, it must not cross over the finish time of the
-    ///   previous segment, or else a std::invalid_argument will be thrown.
+    ///   previous waypoint, or else a std::invalid_argument will be thrown.
     ///
     /// \sa set_finish_time(Time new_time)
-    void adjust_finish_times(Duration delta_t);
-
-    /// Compute the motion across this Trajectory segment
-    std::unique_ptr<Motion> compute_motion() const;
+    void adjust_times(Duration delta_t);
 
     class Implementation;
   private:
 
     /// \internal Private constructor. Use Trajectory::add_segment() to create
-    /// a new Trajectory Segment.
-    Segment();
-    Segment(const Segment&) = delete;
-    Segment(Segment&&) = default;
-    Segment& operator=(const Segment&) = delete;
-    Segment& operator=(Segment&&) = default;
+    /// a new Trajectory Waypoint.
+    Waypoint();
+    Waypoint(const Waypoint&) = delete;
+    Waypoint(Waypoint&&) = default;
+    Waypoint& operator=(const Waypoint&) = delete;
+    Waypoint& operator=(Waypoint&&) = default;
     friend class Trajectory;
     rmf_utils::impl_ptr<Implementation> _pimpl;
   };
 
   // These classes allow users to traverse the contents of the Trajectory.
   // The trajectory operates much like a typical C++ container, but only for
-  // Trajectory::Segment information.
+  // Trajectory::Waypoint information.
   template<typename SegT>
   class base_iterator;
-  using iterator = base_iterator<Segment>;
-  using const_iterator = base_iterator<const Segment>;
+  using iterator = base_iterator<Waypoint>;
+  using const_iterator = base_iterator<const Waypoint>;
 
   /// Create a Trajectory that takes place on the specified map
```

Comment:
Given that a trajectory is now decoupled from any map
```suggestion
  /// Create an empty Trajectory
```